### DISCUSSION TITLE
Server/client vs splits

### DISCUSSION THREAD

**Discussion by KnorrFG** [ki-editor/ki-editor#441] [Ideas]
My main editor is Helix, and one of the things I dislike is to copy and paste stuff between different instances through the system clipboard, and not being able to access shared buffers. It would be nice, if you open ki and no server is running yet, it started a server in the background and connects to it and all further instances then connect to the server, and have access to the same runtime data (buffers, editor clipboards, language server instances, etc.). Then we could simply use whatever we prefer for split management be it zellij, wezterm, i3, sway or something else, and don't need two separate sets of hotkeys for the same thing.

This is also the way Kakoune does it, I think.

Just leaving that here, because splits aren't implemented yet.

EDIT: Maybe you only connect to a server, if its working directory is an ancestor of the working directory of the new client, and otherwise start a new server.

**Comment by wongjiahau** [comment]
Oh wait, so you’re saying if the client-server architecture is implemented like that of Kakoune, window-splitting is no longer a required feature?

**Reply by KnorrFG** [reply]
Well, depends on who you ask, but I wouldn't use it anymore, except maybe for very short-lived splits like jumping to a definition in a new split, reading it, maybe scrolling around some, and then close it again. 

EDIT: I guess Ideally you'd have both, but if I have to choose, I'd go for server/client
